google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -1620,7 +1620,7 @@ public Integer run(DatastoreReaderWriter transaction) {
       datastore.runInTransaction(callable2, readOnlyOptions);
       fail("Expecting a failure");
     } catch (DatastoreException expected) {
-      assertEquals(3, ((DatastoreException) expected.getCause()).getCode());
```

Comment:
It looks like `assertDatastoreException` checks for `expected.getCode()`, but this statement checks for `((DatastoreException) expected.getCause()).getCode()`. If these two methods return different things then should we leave this assert statement here? If these two methods are the same thing then why does the expected code change from 3 to 0?

Reply:
GREAT catch! I've fixed this, PTAL